src/main/java/com/marianbastiurea/lifeofbees/Hive.java

```diff
@@ -515,4 +460,28 @@ public void moveAnEggsFrameFromUnsplitHiveToASplitOne() {
     public void changeQueen() {
         queen = new Queen(0);
     }
-}
\ No newline at end of file
+
+    public void fillUpEggsFrame(Date currentDate, int numberOfEggs) {
+
+        int maxEggPerFrame = 6400; // a frame have around 8500 cells. 75% more or less are used by the queen to lay eggs.
+        // Remaining cells are fill up with honey or are damaged
+
+        int size = getEggsFrames().size() - this.getNumberOfFullEggsFrame();
+        //int remainingEggs = numberOfEggs % size;
+
+        if (size != 0) {
+            int numberOfEggsToPutInFrame = numberOfEggs / size;
+
+            for (EggsFrame eggsFrame : getEggsFrames()) {
+                if (eggsFrame.getNumberOfEggs() < maxEggPerFrame) {
```

Comment:
Move this logic to EggsFrame class in the addEggs method.

Reply:
done

Reply:
This is not done